### ISSUE TITLE
Support client certificate authentication for Smart Card

### ISSUE THREAD

**Issue by arunsingh0708** [a85/POSTMan-Chrome-Extension#3789]
I am using Postman Native app.How can I use desktop Postman when client certificate is on smart card? I can not point to a crt or pfx file, since private key will never leave smart card. What are the options for this?

**Comment by SamvelRaja** [comment]
Thanks for bringing your concern here, we shall analyze the possibilities and provide a solution for this. At present, providing the key file is the only option we have.

**Comment by pvdev** [comment]
Hi,
Just wondering if there has been any progress here. I'd be interested in this solution also.  There does seem to have been strides with node.js and pkcs11. My hope would be an interface similar to FireFox where you can attach to a PKCS11 interface and thus pick up any hardware crypto.  HSPD-12 and CAC included.

Thanks!

**Comment by senseiweb** [comment]
Hey guys...checking the pulse on this particular issue...now that the Chrome version is showing as deprecated, is there any current workarounds or implementations to allow the use of Smart Card authentication.

**Comment by codenirvana** [comment]
@pvdev @senseiweb We have started working towards adding support for PFX certificate, for updates follow this thread #2494

**Comment by RandyAvis** [comment]
Last comment I saw was from February. Still needing SMART CARD support, not just PFX. Need to be able to pull certificate from a card.  Please don't kill the Google app until you have this working.

**Comment by pvdev** [comment]
Yes, PFX is a file format.  PKCS#11 is the standard for interfacing with hardware type devices. There is some work going on. See [PKCS11.js](https://www.npmjs.com/package/pkcs11js). Once your PKCS11 interface is in place you should be able to use the associated libraries from vendors like ActivIdenity or open source like [OpenSC](https://github.com/OpenSC/OpenSC/wiki).

**Comment by shimonacarvalho** [comment]
Any word on smart card support?

**Comment by juanjozorry** [comment]
Please this is a feature we currently need. We can not stop using the Chrome app until the Smart Card feature is implemented in the desktop application.

**Comment by juanjozorry** [comment]
Hi again. A teammate is installing a new development machine and the Crome app is not available to install. Please we need smartcard support on the desktop app. Thank you for your support.

**Comment by cnizzardini** [comment]
Use "Authorize using browser" option. This will open in your browser which should support PIV authentication.

**Comment by micheljung** [comment]
"Authorize using browser" only exists for OAuth

**Comment by ThatTallProgrammer** [comment]
> "Authorize using browser" only exists for OAuth

Can confirm. My team and I are unable to find a way to use smartcards with this system. Is there any update to this issue?

**Comment by rogeruiz** [comment]
Hey folks, I'm curious if there is a better place to track progress on this feature? 

I am currently researching other solutions using the Postman SDK, but am still getting familiar with Postman. I'm set to watch this Issue, but if there is another place I should be listening for updates on this topic, please let me know. Thank you. 

cc: @postman-support

**Comment by franchyze923** [comment]
I'm in need of a way to use a smart card with Postman as well.

**Comment by rogeruiz** [comment]
Thanks @codenirvana it seems I missed this earlier comment but PFX support exists in Postman. So I should be able to solve my issues. I'll post something in this thread about how to do it if that helps future folks, but I think closing this ticket might be a good idea for future folks digging around Support requests. 

👀 https://github.com/postmanlabs/postman-app-support/issues/2494#issuecomment-485894474

**Comment by aliciadavis** [comment]
I need a way to select a certificate from my smart card with postman. I cannot export the private key. I need a way to be prompted to select my certificate.

**Comment by pelesfire19** [comment]
Is there any update to this? Still needing to allow certificate on smart card for API authentication testing

**Comment by Jipwox** [comment]
the PFX certificate file is not a solution to this issue. When performing LDAP / Active Directory auth testing you can't just use a self signed cert file you make yourself if you are testing against a service you don't own. The ability to pass along a smart card's cert is critical for testing especially against military / DoD systems. Please make this a higher priority.

**Comment by adamkorynta** [comment]
> the PFX certificate file is not a solution to this issue. When performing LDAP / Active Directory auth testing you can't just use a self signed cert file you make yourself if you are testing against a service you don't own. The ability to pass along a smart card's cert is critical for testing especially against military / DoD systems. Please make this a higher priority.

Agreed

**Comment by richfergus** [comment]
Yes.. I am forced to build my own tools @postman-staff Any chance you can at least tell us if this is a possibility?

**Comment by ale-rinaldi** [comment]
As a workaround, you may consider using https://github.com/porech/pkcs11-web-proxy to play locally with an API behind PKCS#11 authentication with any tool you like without thinking about authentication.
